src/main/java/edu/ucsb/cs156/courses/controllers/PersonalSchedulesController.java

```diff
@@ -159,6 +159,15 @@ public PersonalSchedule putScheduleById(
             .findByIdAndUser(id, currentUser)
             .orElseThrow(() -> new EntityNotFoundException(PersonalSchedule.class, id));
 
+    Optional<PersonalSchedule> duplicateCheck =
+        personalscheduleRepository.findByUserAndNameAndQuarter(
+            currentUser, incomingSchedule.getName(), incomingSchedule.getQuarter());
+
+    if (duplicateCheck.isPresent()) {
+      throw new IllegalArgumentException(
+          "A personal schedule with that name already exists in that quarter");
+    }
```

Comment:
The problem with this logic is that it is finding the record that *already exists* with this name, which is the record that is actually being edited.

Rather than looking up with an Optional<PersonalSchedule> I think perhaps you need to look up (when doing a PUT) with an Iterable<PersonalSchedule> or List<PersonalSchedule> and check to see if the count is >1.    

This check works for creating a new record, but for editing an existing record, it introduces a bug where it becomes impossible to update the description.